homeassistant/components/volumio/browse_media.py

```diff
@@ -0,0 +1,157 @@
+"""Support for media browsing."""
+import json
+
+from homeassistant.components.media_player import BrowseError, BrowseMedia
+from homeassistant.components.media_player.const import (
+    MEDIA_CLASS_ALBUM,
+    MEDIA_CLASS_ARTIST,
+    MEDIA_CLASS_CHANNEL,
+    MEDIA_CLASS_DIRECTORY,
+    MEDIA_CLASS_GENRE,
+    MEDIA_CLASS_PLAYLIST,
+    MEDIA_CLASS_TRACK,
+    MEDIA_TYPE_MUSIC,
+)
+
+PLAYABLE_ITEM_TYPES = [
+    "folder",
+    "song",
+    "mywebradio",
+    "webradio",
+    "playlist",
+    "cuesong",
+    "remdisk",
+    "cuefile",
+    "folder-with-favourites",
+    "internal-folder",
+]
+
+NON_EXPANDABLE_ITEM_TYPES = [
+    "song",
+    "webradio",
+    "mywebradio",
+    "cuesong",
+    "album",
+    "artist",
+    "cd",
+    "play-playlist",
+]
+
+
+def _item_to_children_media_class(item, info=None):
+    if info and "album" in info and "artist" in info:
+        return MEDIA_CLASS_TRACK
+    if item["uri"].startswith("playlists"):
+        return MEDIA_CLASS_PLAYLIST
```

Comment:
A directory of playlists should have media class directory and children media class playlist.

Reply:
That's exactly what's happening here. Notice that this method determines the children media class.

Reply:
This function is also used inside `_item_to_media_class` which is used to set the media class, not the children media class attribute.

Reply:
Yes, but there it's called with the parent item , not with the current item (so if the parent had `MEDIA_CLASS_PLAYLIST` as the children media class, and we can't determine the media class, we set it to `MEDIA_CLASS_PLAYLIST`).

Reply:
Ok.